### ISSUE TITLE
"pgaudit extension initialized" log entry every 30 sec

### ISSUE THREAD

**Issue by JMLessard** [pgaudit/pgaudit#34]
I am running postgres under windows and get that log entry repeating over and over under a different session id (process id).
It looks like a deamon wakes up every 30 sec and span a new process.
What is this? Any config that I can do to stop that?
Please find bellow an extract of pg_log with the session id between brackets.
`...`
`2017-05-04 09:48:02.670 EDT [590b3112.668] LOG:  pgaudit extension initialized`
`2017-05-04 09:48:02.670 EDT [590b3112.81c] LOG:  pgaudit extension initialized`
`2017-05-04 09:48:02.692 EDT [590b3112.478] LOG:  pgaudit extension initialized`
`2017-05-04 09:48:02.694 EDT [590b3112.b1c] LOG:  pgaudit extension initialized`
`2017-05-04 09:48:19.724 EDT [590b3123.83c] LOG:  pgaudit extension initialized`
`2017-05-04 09:48:19.728 EDT [590b3123.83c] user=[unknown],db=[unknown],app=[unknown],err=00000: LOG:  connection received: host=172.30.33.130 port=53018` 
`2017-05-04 09:48:25.682 EDT [590b3129.b5c] LOG:  pgaudit extension initialized` 
`2017-05-04 09:48:25.686 EDT [590b3129.b5c] user=[unknown],db=[unknown],app=[unknown],err=00000: LOG:  connection received: host=172.30.33.130 port=53019` 
`2017-05-04 09:48:25.690 EDT [590b3129.b5c] user=ftier3,db=ibis,app=[unknown],err=00000: LOG:  connection authorized: user=ftier3 database=ibis` 
`2017-05-04 09:49:32.742 EDT [590b316c.b6c] LOG:  pgaudit extension initialized`
`2017-05-04 09:50:02.840 EDT [590b318a.4b8] LOG:  pgaudit extension initialized`
`2017-05-04 09:50:32.841 EDT [590b31a8.b40] LOG:  pgaudit extension initialized`
`2017-05-04 09:51:02.941 EDT [590b31c6.6e8] LOG:  pgaudit extension initialized`
`2017-05-04 09:51:32.842 EDT [590b31e4.80c] LOG:  pgaudit extension initialized`
`2017-05-04 09:52:02.941 EDT [590b3202.11c] LOG:  pgaudit extension initialized`
`2017-05-04 09:52:32.943 EDT [590b3220.91c] LOG:  pgaudit extension initialized`
`...`

**Comment by jconway** [comment]
This issue was fixed as of this commit: https://github.com/pgaudit/pgaudit/commit/1c99eb969d28e950f8a236bc066aceac169bdc3a
What release version are you using?

**Comment by JMLessard** [comment]
I am running the distribution of bigsql
C:\Program Files\PostgreSQL>pgc list
```
Category     | Component           | Version  | Stage | ReleaseDt  | Status    | Cur? | Updates
PostgreSQL     pg96                  9.6.2-2            2017-03-23   Installed   1
Extensions     pgaudit11-pg96        1.1.0-1            2016-12-28   Installed   1
```

**Comment by dwsteele** [comment]
You'll want to contact OpenSCG with this issue.  They may be building from the wrong version.

**Comment by sravan-velagandula** [comment]
The fix 1c99eb9 exists starting pgaudit-1.0.4. I have tried with pgaudit-1.0.4 and with pgaudit-1.1.0 on windows and was able to reproduce the issue. In fact I added some logging around the code introduced as a part of this fix and it seems the fix does not work.

**Comment by jconway** [comment]
On Windows, shared_preload_libraries does not work the same as on Linux systems. Specifically see the note under: https://www.postgresql.org/docs/9.5/static/runtime-config-client.html#GUC-SHARED-PRELOAD-LIBRARIES

To wit:
   "Note: On Windows hosts, preloading a library at server start will
    not reduce the time required to start each new server process; each
    server process will re-load all preload libraries. However,
    shared_preload_libraries is still useful on Windows hosts for
    libraries that need to perform operations at postmaster start time."

So I would expect to see that initialization message once every time a client connects and there is nothing we can do about it on Windows short of removing the message entirely, or maybe demoting it to a debug message.

Since you are seeing it every 30 seconds I would guess that it is the either autovacuum daemon launching or possibly something you are doing from your app or some other scheduled task. I don't have Windows handy to test out the theory, but you could try changing autovacuum_naptime and see if the interval between messages changes accordingly.

**Comment by JMLessard** [comment]
The message is effectively coming from autovacumm. I changed the naptime to 5min and the message appears every 2min30sec (the cluster is runnig two databases). The best would be removing the message entirely.

**Comment by jconway** [comment]
Resolved with commit 9185dec. On EXEC_BACKEND builds (e.g. Windows) the message is now demoted to level DEBUG1.
